Subbu/goal generation

SubramanianKrish opened the pull request:
Realsense namespace corrected in sim
static map params yaml removed and moved to gloobal params file in both sim and hw
depth registered on rgb topic
goal generation package created (couple of nodes, couple of launch files)
Done upto projecting lane lines in pcl version onto the costmap using movebase obstacle layer plugin

Commit:
Changed the ground plane to be artificially green
Added online segmentation map generating script

Added the goal generation package
Corrected the namespace for realsense in simulation

Added lane segmentation in cloud

Added the goal generation package
Corrected the namespace for realsense in simulation

Added lane segmentation in cloud

Added lane segmenter in cloud to project the segmented depth map of the lane back onto the real world
Added a fake "lane" sensor in the costmap commons params file

/realsense namespace removed
depth image registered to rgb image
lane pcl projected onto constmap
goal transfer launch files created

removed static params yaml and updated global costmap params

removed static map params and updated global cmap params for sim and hw

Changed the ground plane to be artificially green
Added online segmentation map generating script

Added lane segmentation in cloud

segmap uncommented in launch file

kob51 commented on autovalet_gazebo/models/cmu_ground_plane/materials/scripts/cmu_ground_plane.material:
```diff
@@ -10,7 +10,7 @@ material CMUGroundPlane/Image
             texture_unit
             {
-                texture dark-garage_onlyRightCurve.jpg
+                texture garage_green.jpg
```
don't push this change to master.

kob51 commented on autovalet_goal_generation/launch/seg_map.launch:
```diff
@@ -0,0 +1,85 @@
+<?xml version="1.0"?>
```
is this file not a lot of copy/pasted from parking_lot.launch? could we clean that up?

SubramanianKrish replied:
Yeah. it's mostly a copy of parking_lot launch. How do you want me to clean it up?

kob51 commented on autovalet_goal_generation/scripts/transfer_lane.py:
```diff
@@ -0,0 +1,131 @@
+    self.ats.registerCallback(self.callback)
+
+    # Define the camera matrix individual values for projection
+    self.fx     = self.camera.K[0]
```
where's the 5th dof of K?

evamo0508 replied:
I believe there are only 4: cx, cy, fx, fy

SubramanianKrish replied:
skew is assumed zero

kob51 commented on autovalet_goal_generation/scripts/transfer_lane.py:
```diff
@@ -0,0 +1,131 @@
+    lane_lines = cv2.dilate(lane_lines,self.dilation_kernel,iterations = 1)
+
+    # Overlay mask
+    lane_depth_image = cv2.bitwise_and(depth_image, depth_image, mask = lane_lines/255)
```
is "depth_image" both src and destination? wondering why it comes up twice in parameters

SubramanianKrish replied:
I used this as a reference .. didn't dig too deep there
https://docs.opencv.org/master/d0/d86/tutorial_py_image_arithmetics.html

kob51 commented on autovalet_goal_generation/scripts/transfer_lane.py:
```diff
@@ -0,0 +1,131 @@
+	x = (u.flatten() - self.cx)/self.fx
+	y = (v.flatten() - self.cy)/self.fy
+
+	z = depth_image.flatten() / 1000.0;
```
so you normalize Z then multiply X and Y by Z?

evamo0508 replied:
the default z values come in `mm`, but we want meters, that's why we divided them by 1000.0

SubramanianKrish replied:
The plugin gives the output in mm. Similar to actual camera if i'm not wrong

kob51 replied:
i get the metric conversion, now look at lines 105 and 106. multiplying x and y by z

evamo0508 replied:
that's a part of the camera coordinate transformation stuff which @SubramanianKrish  and I had gone through the derivations the other day. just think of the `s[u v 1] = KD[x y z 1]` equation and mess around with the three separate equations for each element, you'll get it.

kob51 commented on autovalet_navigation/config/husky/static_map_params.yaml:
```diff
@@ -1,2 +0,0 @@
-map_topic: "/rtabmap/grid_map"
```
why did you delete these files (same one in husky and sim dirs)? where does this info live now?

SubramanianKrish replied:
file cleanup. Neater that way

kob51 commented on autovalet_navigation/config/sim/global_costmap_params.yaml:
```diff
@@ -1,13 +1,24 @@
       - {name: inflation,      type: "costmap_2d::InflationLayer"}
+      - {name: lane,           type: "costmap_2d::ObstacleLayer"}
+
+  lane:
```
why does this not live in the costmap_common_params.yaml file next to the other pcl sources?

SubramanianKrish replied:
It didn't work if I put it in costmap common. Had to be here for it to work

kob51 commented on autovalet_navigation/config/sim/global_costmap_params.yaml:
```diff
@@ -1,13 +1,24 @@
\ No newline at end of file
+  map_type: costmap
+
+  static_layer:
```
oh you moved it here. make sure you do the same thing on the `husky` version of this file so we don't forget about it.

SubramanianKrish replied:
Yup. Done.

mishra-nikita reviewed (dismissed):
kewlls

evamo0508 commented on autovalet_gazebo/launch/valet_rtab_sim.launch:
```diff
@@ -47,7 +47,7 @@
         <param name="Icp/PointToPlane"        type="string" value="true"/>
         <param name="Icp/Iterations"          type="string" value="20"/>
-        <param name="Icp/VoxelSize"           type="string" value="0.01"/> <!-- 0.01 works pretty well -->
+        <param name="Icp/VoxelSize"           type="string" value="0.1"/> <!-- 0.01 works pretty well -->
```
why is this? Shouldn't increasing voxel size lower the localization accuracy? or did we want to prevent some errors happening due to small voxelSize?

SubramanianKrish replied:
Oh that's to prevent all the error messages about leaf size coming up on the terminal. Sure smaller voxels are more accurate but we thought they may be eating up compute at the planner stage. This was one of the experimentation changes. Don't know how it made through to the PR but I think we can keep it as it removes the warning messages on the slam terminal

evamo0508 reviewed (commented):
Why are there two `generate_gt_segmap.py`?

SubramanianKrish committed:
Delete generate_gt_segmap.py

This seems to be a ghost file in lane_detection package which is now legacy. Don't know how but seems to have been created in one of the patches. I had it on my local machine too.. it's not needed in the master. So deleting it!

SubramanianKrish commented:
> Why are there two `generate_gt_segmap.py`?

I've deleted the ghost file created under lane_detection. It should not be in the PR anymore

SubramanianKrish committed:
Update cmu_ground_plane.material

reverted to original texture to prevent any sim world issues.

kob51 commented on autovalet_goal_generation/launch/seg_map.launch:
```diff
@@ -0,0 +1,85 @@
+<!--
```
so it seems like a lot of the calls in this file could be separated out into its own file. and thne maybe both parking_lot.launch and this file could call that file? idk. could be more trouble when it's worth, i just don't love the idea of copy-pasting code. your call

SubramanianKrish replied:
Done. Check it out now!

Commit:
created a new green world. Yeah you heard me right
